lib/gooddata/bricks/middleware/execution_result_middleware.rb

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2019, GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_middleware'
+
+module GoodData
+  module Bricks
+    module ExecutionStatus
+      OK = 'OK'
+      ERROR = 'ERROR'
+      WARNING = 'WARNING'
+    end
+
+    class ExecutionResultMiddleware < Bricks::Middleware
+      def call(params)
+        params = self.class.update_execution_result(params, params["EXECUTION_RESULT_STATUS"], params["EXECUTION_RESULT_MESSAGE"])
+        @app.call(params)
+      end
+
+      # Update process execution result when the script is executed inside a GDC ETL process execution (ruby bricks).
+      # Ruby bricks should update execution result at the end of script with status=WARNING or OK and summary message if the script exits normally.
+      # If any fatal error, script should update execution result with status=ERROR and error message,
+      # then throw exception to notify GDC platform that the script exists unexpectedly.
+      # @param [HashMap] params contains GDC_EXECUTION_RESULT_LOG_PATH or GDC_LOG_DIRECTORY, GDC_EXECUTION_ID
+      # @param [ExecutionStatus] status execution status
+      # @param [String] message execution message
+      def self.update_execution_result(params, status, message = "")
+        if status != ExecutionStatus::OK && status != ExecutionStatus::ERROR && status != ExecutionStatus::WARNING
+          GoodData.logger.warn("Unknown execution status #{status}, ignored it.")
+        end
+
+        result = {
+          executionResult: {
+            status: status,
+            message: message
+          }
+        }
+        update_result(params, result)
+      end
+
+      private
+
+      def self.update_result(params, result)
+        log_directory = params['GDC_LOG_DIRECTORY']
+        execution_id = params['GDC_EXECUTION_ID']
+        execution_result_logger_file = params['GDC_EXECUTION_RESULT_LOG_PATH'] || ENV['GDC_EXECUTION_RESULT_LOG_PATH']
+        execution_result_logger_file = execution_result_logger_file.nil? ? "#{log_directory}/#{execution_id}_result.json" : execution_result_logger_file
```

Comment:
log_directory and execution_id are passed by k8s executor manager only, so please check nil for backward compatibility
it should be: 
`execution_result_logger_file = "#{log_directory}/#{execution_id}_result.json" if execution_result_logger_file.nil? && !log_directory.nil?`

Reply:
Yes, I fixed